rust_src/src/composite.rs

```diff
@@ -0,0 +1,38 @@
+//! Composite sequence support.
+
+use libc::ptrdiff_t;
+
+use remacs_macros::lisp_fn;
+
+use crate::{
+    lisp::LispObject,
+    remacs_sys::{compose_text, validate_subarray, CHECK_STRING, SCHARS},
+};
+
```

Comment:
Normally, the Remacs style is to use LispString and EmacsInt instead of LispObject. However, since all of the calls are still in C there is little value to using the new names at the moment.

Both `validate_subarray` and `compose_text` appear to be straight forward to port. Would you be up to giving them a go? Then this function could get a proper Rust port.

For `validate_subarray`, make a Rust function with the following signature:

`pub fn validate_subarray_rust(array: LispObject, from: EmacsInt, to: EmacsInt, size: isize) -> (EmacsInt, EmacsInt)`
then make one with the C signature:
```
#[no_mangle]
pub extern "C" fn validate_subarray(array: LispObject, from: LispObject, to: LispObject, size: isize, new_from: &mut LispObject, new_to: &mut LispObject) // notice there is no return this time
```

The idea being we make the Rust functions behave in an idiomatically Rust fashion and wrap them to the keep the original C working.

Reply:
sure, and thank you for the advice.

Reply:
The original function would turn nil into 0, is it okay if I make from and to LispObjects, and then convert them to EmacsInt inside the function to check for nil?